src/uct/base/uct_component.h

```diff
@@ -179,6 +179,58 @@ struct uct_component {
     UCS_CONFIG_REGISTER_TABLE_ENTRY(&(_component)->cm_config, &ucs_config_global_list);
 
 
+/**
+ * Call component registration routine.
+ *
+ * @param [in] _name  Component name
+ */
+#define uct_component_register(_name) \
+    uct_component_##_name##_ctor()
+
+
+/**
+ * Call component unregistration routine.
+ *
+ * @param [in] _name  Component name
+ */
+#define uct_component_unregister(_name) \
+    uct_component_##_name##_dtor()
+
+
+/**
+ * Declare component registration routines.
+ * @ref uct_query_components.
+ *
+ * @param [in] _name Component name used to generate ctors/dtors.
+ * @param [in] ...   Routine attributes (static)
+ */
+#define UCT_COMPONENT_REGISTER_DECL(_name, ...) \
+    __VA_ARGS__ void uct_component_##_name##_ctor(void); \
+    __VA_ARGS__ void uct_component_##_name##_dtor(void);
+
+
+/**
+ * Component registration routines.
+ * @ref uct_query_components.
+ *
+ * @param [in] _component  Pointer to a global component structure to register.
+ * @param [in] _name       Component name used to generate ctors/dtors.
+ */
+#define UCT_COMPONENT_REGISTER_DEF(_component, _name) \
+    extern ucs_list_link_t uct_components_list; \
+    void uct_component_##_name##_ctor(void) \
+    { \
+        ucs_list_add_tail(&uct_components_list, &(_component)->list); \
+        ucs_list_add_tail(&ucs_config_global_list, &(_component)->md_config.list); \
+        ucs_list_add_tail(&ucs_config_global_list, &(_component)->cm_config.list); \
+    } \
+    void uct_component_##_name##_dtor(void) \
+    { \
+        ucs_list_del(&(_component)->md_config.list); \
```

Comment:
why remove from config list ut not from the components list?
pls add TODO comment if needed